iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -621,8 +617,13 @@ private void commitCompaction(Table table, long startTime, FilesForCommit result
       LOG.debug("Compacted full table with files {}", results);
     } else {
       // Single partition compaction
-      List<DataFile> existingDataFiles = IcebergTableUtil.getDataFiles(table, partitionSpecId, partitionPath);
-      List<DeleteFile> existingDeleteFiles = IcebergTableUtil.getDeleteFiles(table, partitionSpecId, partitionPath);
+      List<DataFile> existingDataFiles = partitionPath != null ?
+          IcebergTableUtil.getDataFiles(table, partitionSpecId, partitionPath) :
+          IcebergTableUtil.getDataFilesNotInSpec(table, partitionSpecId);
+
+      List<DeleteFile> existingDeleteFiles = partitionPath != null ?
+          IcebergTableUtil.getDeleteFiles(table, partitionSpecId, partitionPath) :
```

Comment:
should we add extra param like `latestSpecOnly` in IcebergTableUtil.getDataFiles instead. of creating multiple methods?

Reply:
Yes, I added one more parameter to the function and got rid of the overloaded functions.